libraries/AP_GPS/AP_GPS.cpp

```diff
@@ -442,22 +548,12 @@ AP_GPS::detect_instance(uint8_t instance)
 AP_GPS::GPS_Status 
 AP_GPS::highest_supported_status(uint8_t instance) const
 {
-    if (drivers[instance] != nullptr) {
+    if (instance < GPS_MAX_RECEIVERS && drivers[instance] != nullptr) {
```

Comment:
I considered adding the extra logic to do as you suggest - to use the real GPSs actual "highest_supported_status" but at the moment, this method is never called for the blended GPS.  It's only called for the two real instances and it's only used to decide whether we should send RTK status or not to the GCS.  So if I added this logic, it would never actually be used so I decided the additional complexity for no immediate benefit wasn't worth it and instead just added the comments warning about the limitation.